examples/connext_dds/printing_qos/README.md

```diff
@@ -0,0 +1,128 @@
+# Example Code: Print QoS
+
+Example about printing the QoS of the entities.
+
+## Concept
+
+RTI Connext DDS provides you the possibility to print the following Qos objects:
+`DataReaderQos`, `DataWriterQos`, `PublisherQos`, `SubscriberQos`, `TopicQos`,
+`DomainParticipantQos` and `DomainParticipantFactoryQos`.
+
+-   In C, there are three new APIs per top-level QoS object (DataWriterQos is
+ used as an example below):
+
+```plaintext
+DDS_DataWriterQos_print(const struct DDS_DataWriterQos *self)
+DDS_DataWriterQos_to_string(const struct DDS_DataWriterQos *self, char *string, DDS_UnsignedLong *string_size)
+DDS_DataWriterQos_to_string_w_params(const struct DDS_DataWriterQos *self, char *string, DDS_UnsignedLong *string_size, const struct DDS_DataWriterQos *base, const struct DDS_QosPrintFormat *format)
+```
+
+- In Traditional C++, the same functionality is achieved through overloads:
+
+```plaintext
+DDS_DataWriterQos::print()
+DDS_DataWriterQos::to_string(char *string, DDS_UnsignedLong& string_size)
+DDS_DataWriterQos::to_string(char *string, DDS_UnsignedLong& string_size, const DDS_DataWriterQos& base)
+DDS_DataWriterQos::to_string(char *string, DDS_UnsignedLong& string_size, const DDS_QosPrintFormat &format)
+DDS_DataWriterQos::to_string(char *string, DDS_UnsignedLong& string_size, const DDS_DataWriterQos &format, const DDS_QosPrintFormat &format)
+```
+
+- In Modern C++, the to_string APIs are free-standing functions:
+
+```plaintext
+std::string to_string(const DataWriterQos& qos, const QosPrintFormat& format = QosPrintFormat())
+std::string to_string(const DataWriterQos& qos, const DataWriterQos& base, const QosPrintFormat& format = QosPrintFormat())
+std::string to_string(const DataWriterQos& qos, const qos_print_all_t& qos_print_all, const QosPrintFormat& format = QosPrintFormat())
+std::ostream& operator<<(std::ostream& out, const DataWriterQos& qos)
+```
+
+- In Java, Object.toString is overridden, and additional overloads are available:
+
+```plaintext
+String DataWriterQos.toString()
+String DataWriterQos.toString(DataWriterQos baseQos, QosPrintFormat format)
+String DataWriterQos.toString(QosPrintFormat format)
+String DataWriterQos.toString(DataWriterQos baseQos)
+```
+
+- In .NET, Object.ToString is overridden, and additional overloads are available:
+
+```plaintext
+String ^DataWriterQos::ToString()
+String ^DataWriterQos::ToString(DataWriterQos ^base, QosPrintFormat ^format)
+String ^DataWriterQos::ToString(QosPrintFormat ^format)
+String ^DataWriterQos::ToString(DataWriterQos ^base)
+```
+
+For more information about the use of these APIs, please refer to the API
```

Comment:
Provide relevant link to UM (if there isn't one, don't worry -- but I think we wrote some stuff for this)

Reply:
I could not find it in the UM. Can you help me with that :)

Reply:
https://buildlogs.rti.com/atlassian/community/documentation/connext-dds/develop/doc/manuals/connext_dds_professional/users_manual/index.htm#users_manual/QoS_Profile_Inheritance.htm#19.3.3.5_Viewing_Resolved_QoS_Values%3FTocPath%3DPart%25203%253A%2520Advanced%2520Concepts%7C19.%2520Configuring%2520QoS%2520with%2520XML%7C19.3%2520QoS%2520Profiles%7C19.3.3%2520QoS%2520Profile%2520Inheritance%2520and%2520Composition%7C_____5